datafusion-examples/examples/regexp.rs

```diff
@@ -0,0 +1,300 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
```

Comment:
IMHO I like adding examples for introduced functions, but we probably can do that on SQL only? like https://spark.apache.org/docs/latest/api/sql/index.html

Reply:
When I started with DataFusion it was the examples that gave me the initial base to work from - especially since this projects was essentially my first time with Rust code. I personally think having both dataframe and SQL examples for the same thing to be quite helpful for those getting started who are not already 

I was actually thinking of spending some time (when I had it) to enhance the scalar docs to add examples where missing and to add whole new sections to the examples as above.

Reply:
I'm thinking on doc tests? like in `datafusion/core/src/dataframe/mod.rs`

Reply:
That could work though it could be rather large doc. I'll update this PR in the next day or two with that change. I should likely do that with the to_timestamp example as well (in a separate PR of course).

Reply:
Done.